lib/checkmot/client.go

```diff
@@ -32,9 +32,47 @@ func (c *Client) GetRecordByVRM(ctx context.Context, vrm string) (res *Vehicle,
 		return
 	}
 
-	if len(out) != 1 {
-		return nil, errors.New(ErrUnexpectedResultCount)
+	if len(out) == 0 {
+		return nil, errors.New(ErrNoResults)
+	}
+
+	if len(out) == 1 {
+		return out[0], nil
+	}
+
+	return attemptToMergeTestsForTheSameVehicle(out)
+}
+
+// the API sometimes returns multiple vehicles with the same details even for a specific vehicle search
+// merge the list of MOT tests if all vehicle details are the same
+func attemptToMergeTestsForTheSameVehicle(out []*Vehicle) (*Vehicle, error) {
+	for i := 1; i < len(out); i++ {
+		if !isSameVehicle(out[0], out[i]) {
+			return nil, errors.New(ErrMultipleVehicles)
+		}
+
+		out[0].MOTTests = append(out[0].MOTTests, out[i].MOTTests...)
 	}
 
 	return out[0], nil
 }
+
+func isSameVehicle(a *Vehicle, b *Vehicle) bool {
```

Comment:
Is the "first used date" not handy to tell if it's the same vehicle?

Reply:
hmm yeah, shall I add that and manufactureYear, did this a while ago and not sure why I have left those out 🤷

Reply:
Done